packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,389 @@
+  ]),
+
+  /**
+   * The set of precisions that can be used with the onset picker.
+   * One of OnsetPicker.Opts.Precisions.ONAT, OnsetPicker.Opts.Precisions.ABOUT, OnsetPicker.Opts.Precisions.BEFORE,
+   * OnsetPicker.Opts.Precisions.AFTER, OnsetPicker.Opts.Precisions.UNKNOWN.
+   * Order of precisions determines order in precision select.
+   */
+  precisionSet: PropTypes.arrayOf(PropTypes.oneOf([
+    OnsetUtils.PrecisionOptions.ONAT,
+    OnsetUtils.PrecisionOptions.ABOUT,
+    OnsetUtils.PrecisionOptions.BEFORE,
+    OnsetUtils.PrecisionOptions.AFTER,
+    OnsetUtils.PrecisionOptions.UNKNOWN,
+  ])),
+
+  /**
+   * Name of the precision select. The name should be unique.
+   */
+  precisionSelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a precision is selected.
+   * The first parameter is the changed precision value.
+   */
+  precisionSelectOnChange: PropTypes.func,
+
+  /**
+   * The ISO 8601 string representation of the onset date to view/modify. Defaults to current date.
+   */
+  onsetDate: PropTypes.string,
+
+  /**
+   * Name of the onsetDate input. The name should be unique.
+   */
+  onsetDateInputName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a onsetDate is changed.
+   * The first parameter is the changed onsetDate value.
+   */
+  onsetDateInputOnChange: PropTypes.func,
+};
+
+const defaultProps = {
+  birthdate: undefined,
+  granularity: 'month',
+  granularitySelectName: undefined,
+  granularitySelectOnChange: undefined,
+  precision: 'on/at',
+  precisionSet: [
+    'on/at',
+    'about',
+    'before',
+    'after',
+    'unknown',
+  ],
+  precisionSelectName: undefined,
+  precisionSelectOnChange: undefined,
+  onsetDate: undefined,
+  onsetDateInputName: undefined,
+  onsetDateInputOnChange: undefined,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class OnsetPicker extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    // This binding is necessary to make `this` work in the callback
+    this.changeGranularity = this.changeGranularity.bind(this);
+    this.changePrecision = this.changePrecision.bind(this);
+    this.changeAge = this.changeAge.bind(this);
+    this.changeAgeUnit = this.changeAgeUnit.bind(this);
+    this.changeYear = this.changeYear.bind(this);
+    this.changeMonth = this.changeMonth.bind(this);
+    this.changeDate = this.changeDate.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+
+    const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, this.props.onsetDate);
+
+    this.state = {
+      granularity: this.props.granularity,
+      precision: this.props.precision,
+      onsetDate: this.props.onsetDate ? this.props.onsetDate : moment(),
```

Comment:
Based on how this.state.onsetDate is used throughout this file, it is an ISO 8601 string. However, if this.props.onsetDate is null, this.state.onsetDate would end up being a moment object by calling moment().

Reply:
Now converting the onsetDate prop to a moment object if supplied to be consistent. The moment object is only used internally and a ISO 8601 is returned to the user any time a update to the onsetDate occurs.